main.tf

```diff
@@ -0,0 +1,170 @@
+}
+
+resource "aws_iam_role_policy" "log_agent" {
+  name   = "${var.name_prefix}-log-permissions"
+  role   = "${aws_iam_role.task.id}"
+  policy = "${data.aws_iam_policy_document.task_permissions.json}"
+}
+
+# ------------------------------------------------------------------------------
+# Security groups
+# ------------------------------------------------------------------------------
+resource "aws_security_group" "ecs_service" {
+  vpc_id      = "${var.vpc_id}"
+  name        = "${var.name_prefix}-ecs-service-sg"
+  description = "Fargate service security group"
+  tags        = "${var.tags}"
+}
+
+resource "aws_security_group_rule" "ingress_service" {
+  security_group_id = "${aws_security_group.ecs_service.id}"
+  type              = "ingress"
+  protocol          = "icmp"
+  from_port         = "8"
+  to_port           = "0"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "egress_service" {
+  security_group_id = "${aws_security_group.ecs_service.id}"
+  type              = "egress"
+  protocol          = "-1"
+  from_port         = 0
+  to_port           = 0
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+# ------------------------------------------------------------------------------
+# LB Target group
+# ------------------------------------------------------------------------------
+resource "aws_lb_target_group" "task" {
+  vpc_id       = "${var.vpc_id}"
+  protocol     = "${var.task_container_protocol}"
+  port         = "${var.task_container_port}"
+  target_type  = "ip"
+  health_check = ["${var.health_check}"]
+
+  # NOTE: TF is unable to destroy a target group while a listener is attached,
+  # therefor we have to create a new one before destroying the old. This also means
+  # we have to let it have a random name, and then tag it with the desired name.
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  tags = "${merge(var.tags, map("Name", "${var.name_prefix}-target-${var.task_container_port}"))}"
+}
+
+# ------------------------------------------------------------------------------
+# ECS Task/Service
+# ------------------------------------------------------------------------------
+data "null_data_source" "task_environment" {
+  count = "${var.task_definition_environment_count}"
+
+  inputs = {
+    name  = "${element(keys(var.task_definition_environment), count.index)}"
+    value = "${element(values(var.task_definition_environment), count.index)}"
+  }
+}
+
+resource "aws_ecs_task_definition" "task" {
+  family                   = "${var.name_prefix}"
+  execution_role_arn       = "${aws_iam_role.execution.arn}"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = "${var.task_definition_cpu}"
+  memory                   = "${var.task_definition_memory}"
+  task_role_arn            = "${aws_iam_role.task.arn}"
+
+  container_definitions = <<EOF
+[{
+    "cpu":0,
```

Comment:
This looks kind of curious to me:
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html#fargate-task-defs

There is no valid cpu configuration that is `cpu: 0`, that I can see in the docs at least. Does it work? What is the rationale? 😆

Reply:
I think you are confusing cpu on task level and on container level (this part is very tricky). On task level as you said it needs to be one of the valid configurations and for Fargate this field is required. On container level this configures cpu shares, https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_environment. 0 is a valid entry here but since its optional we should be able to remove it completely.